hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamer.java

```diff
@@ -640,10 +640,15 @@ public static void main(String[] args) throws Exception {
       LOG.warn("--enable-hive-sync will be deprecated in a future release; please use --enable-sync instead for Hive syncing");
     }
 
+    int exitCode = 0;
     try {
       new HoodieStreamer(cfg, jssc).sync();
+    } catch (Exception e) {
+      LOG.error("Failed to execute HoodieStreamer : {}", e.getMessage());
+      exitCode = 1;
+      throw e;
     } finally {
-      jssc.stop();
+      jssc.sc().stop(exitCode);
```

Comment:
Thanks for the improvement.  It looks like this API is introduced in Spark 3.4, so it does not compile on Spark 3.3.  Could you adapt your change on that?